core-bundle/src/EventListener/FilterPageTypeListener.php

```diff
@@ -27,21 +27,22 @@ public function __construct(private Connection $connection)
     public function __invoke(FilterPageTypeEvent $event): void
     {
         $dc = $event->getDataContainer();
+        $currentRecord = $dc->getCurrentRecord();
 
-        if (!$dc->activeRecord) {
+        if (null === $currentRecord) {
             return;
         }
 
         // The first level can only have root pages (see #6360)
-        if (!$dc->activeRecord->pid) {
+        if (!($currentRecord['pid'] ?? null)) {
```

Comment:
Why `($currentRecord['pid'] ?? null)` here but only `$currentRecord['pid']` in line 45?

Reply:
Because we need to cover both cases, `isset()` and `pid = 0`. And the second case doesn't need that check because this one here is an early return.